detectors/aws/ecs/ecs.go

```diff
@@ -43,27 +43,32 @@ var (
 )
 
 // Create interface for methods needing to be mocked
-type detectorUtils interface {
+type detectorUtilsResources interface {
 	getContainerName() (string, error)
 	getContainerID() (string, error)
 }
 
-// struct implements detectorUtils interface
-type ecsDetectorUtils struct{}
+// struct implements detectorUtilsResources interface
+type DetectorUtils struct{}
```

Comment:
This should remain unexported and can we revert the rename to narrow the changes in this PR to solely address the bug.